### PR TITLE
Feature/kebechet update manager

### LINKED ISSUE #198: Add `.thoth.yaml` file entity
**Is your feature request related to a problem? Please describe.**
In order to inspect kebechet managers, we need to find out which of them are used in the repository.

**Describe the solution you'd like**
~~Inspect thoth-yaml configuration file and detect managers from there.~~

~~Thoth-yaml Entity acceptance criteria:~~

- [ ] ~~parse all of the managers specified under `- managers:` and save them
Therefore for each repo that uses kebechet, the extracted knowledge would be defined as:~~
~~` {
    <enabledManager1> : {...},
    <enabledManager2> : {...},
    ....
}`~~

~~Where each manager would have own template of knowledge ( the unspecified `{...}` ).~~
(see UPDATE note)

For now, it is not important to implement all of them. Therefore I select only one of them and the other ones will be then added furthermore. 

**_Kebechet `update` manager_**
We have to vary between the manual and automatic approach.

- [x] manual approach
Extracted information: Issue created timestamp, Sesheta first comment timestamp,
then the Time To Register a Request can be calculated in processing -> Δ( t(issue_opened), t(first kebechet message) ) 

- [x] automatic
Check release time of a new python module and compare that with when the kebechet made update pullrequest
Extracted information: Package release version timestamp, Issue created timestamp,

Also other information should be extracted:

- [x]  'Failed to update dependencies to their latest version' Issue
Extracted information: Timestamp

 

**UPDATE:**
I've decided to not implement thoth-yaml entity but rather a KebechetUpdateManager entity. This comes as a more simpler and reasonable approach when it comes to storing json files of Issue entities (as update requests) and then processing them one by one in e.g. Superset.

### PR DESCRIPTION
**Pull request by xtuchyna**
## Related Issues and Dependencies
This PR closes #198 

## This introduces a breaking change

- [ ] Yes
- [X] No

### CODE REVIEW

File: srcopsmetrics/entities/dependency_update.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from typing import Any, List
+
+from github.Commit import Commit
+from voluptuous.schema_builder import Schema
+
+from srcopsmetrics.entities import Entity
+
+
+class DependencyUpdate(Entity):
+    """Template entity.
```
Comment (pacospace) [inline srcopsmetrics/entities/dependency_update.py:29]: specific description?

**Comment by xtuchyna** [comment]
With new api rate limit check, this fixes #130

File: srcopsmetrics/entities/dependency_update.py [hunk 2]
```diff
@@ -26,22 +26,15 @@
 
 
 class DependencyUpdate(Entity):
-    """Template entity.
+    """Python Dependency update entity.
```
Comment (pacospace) [inline srcopsmetrics/entities/dependency_update.py:29]: This class is not part of EnumTypes, so it cannot be used with `-e` in CLI

Comment (xtuchyna) [reply srcopsmetrics/entities/dependency_update.py:29]: not needed anymore, see this https://github.com/thoth-station/mi/pull/186#discussion_r501320326

Comment (pacospace) [reply srcopsmetrics/entities/dependency_update.py:29]: But if you try running with `-e` the CLI you cannot select the entities you want now

Comment (pacospace) [reply srcopsmetrics/entities/dependency_update.py:29]:
```
pipenv run python3 srcopsmetrics/cli.py --repository thoth-station/mi -c -l -e DependencyUpdate 
Traceback (most recent call last):
  File "srcopsmetrics/cli.py", line 26, in <module>
    from srcopsmetrics.bot_knowledge import analyse_projects
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/bot_knowledge.py", line 27, in <module>
    from srcopsmetrics.entities import Entity, NOT_FOR_INSPECTION
ImportError: cannot import name 'Entity'
```

**Comment by xtuchyna** [comment]
@pacospace pingy ping

File: srcopsmetrics/iterator.py [hunk 3]
```diff
@@ -69,7 +69,7 @@ def run(self):
             for idx, entity in enumerate(self.entity.analyse(), 1):
                 self.knowledge_updated = True
 
-                remaining = github.rate_limiting[0]
+                remaining = github.get_rate_limit().core
```
Comment (pacospace) [inline srcopsmetrics/iterator.py:72]: this change produced errors: 

```
INFO:srcopsmetrics.bot_knowledge:########################
INFO:srcopsmetrics.bot_knowledge:DependencyUpdate inspection
INFO:srcopsmetrics.entities.tools.storage:Loading knowledge from Ceph
INFO:srcopsmetrics.entities.interface:No previous knowledge of type DependencyUpdate found
INFO:srcopsmetrics.iterator:-------------DependencyUpdate Analysis-------------
INFO:srcopsmetrics.iterator:Cached knowledge could not be saved
Traceback (most recent call last):
  File "srcopsmetrics/cli.py", line 129, in <module>
    cli()
  File "/home/fmurdaca/.local/share/virtualenvs/mi-_yZ92-Ga/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/fmurdaca/.local/share/virtualenvs/mi-_yZ92-Ga/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/fmurdaca/.local/share/virtualenvs/mi-_yZ92-Ga/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/fmurdaca/.local/share/virtualenvs/mi-_yZ92-Ga/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "srcopsmetrics/cli.py", line 111, in cli
    analyse_projects(projects=tupled_repos, is_local=is_local, entities=entities)
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/bot_knowledge.py", line 90, in analyse_projects
    github_repo=github_repo, project_path=project_path, entity_cls=entity, is_local=is_local
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/github_knowledge.py", line 187, in analyse_entity
    analysis.run()
  File "/home/fmurdaca/work/aicoe/mi/srcopsmetrics/iterator.py", line 74, in run
    if remaining <= API_RATE_MINIMAL_REMAINING:
TypeError: '<=' not supported between instances of 'Rate' and 'int'
```

Comment (xtuchyna) [reply srcopsmetrics/iterator.py]: nice catch!
i fixed it so it is now relevant with github utc time, too

**Review by pacospace — APPROVED** [review]
Thanks! :100: :1st_place_medal:

**Comment by xtuchyna** [comment]
with https://github.com/thoth-station/mi/pull/251/commits/98e7db6a75a6e644041b7264a550d638594e2c3c commit, this PR fixes #130

**Pull request merged**